README.md

```diff
@@ -177,6 +178,8 @@ You can limit the maximum rank by setting ``-DMAXRANK=<num>`` in the ``FYPPFLAGS
 make -f Makefile.manual FYPPFLAGS=-DMAXRANK=4
 ```
 
+You can edit the compiler and its optimization flags by specifying the `FC` and `FFLAGS` variables in Makefile.manual. The default `FFLAGS` does not include any optimization, and you may want to add `-O3 -flto` to its definition. 
```

Comment:
Editing makefiles shouldn't be encouraged here, instead can we modify the makefile to accept the environment variables as default? Something along the lines of:

```make
FC ?= gfortran
FFLAGS ?= ...
```

The manual makefile would be used with

```
make -f Makefile.manual FFLAGS="-O3 -flto"
```

Reply:
Agree that's a nice way of doing it. 

The only potential issue I can think of: Suppose a user accidently has `FC` or `FFLAGS` already defined, and are not aware of that-- then they might get confusing results. 

But if the README encourages users to define those two variables, then I doubt that will be a problem.

Reply:
I'm hearing this argument a lot, how do you set `FC` or `FFLAGS` by accident? Usually those variables are set by environment modules which load compilers, where else?

Reply:
A situation would be where a user has a script that they know can be used to setup the environment for some software -- but they don't know/understand the contents of the script.

For example the [clawpack software installation instructions](https://www.clawpack.org/installing.html) encourage users to define: 

```
export CLAW=$HOME/clawpack_src/clawpack-v5.8.0
export FC=gfortran
```

I have a script "setup_clawpack.sh" in my home directory that does this. If using clawpack, I would simply source the script and then do stuff, but not understand how my environment was modified. And similarly for other software.

Obviously users with enough experience will appreciate the risks here, but I suspect many users do not.

Reply:
I am bit confused. I thought at the end the main use case will be that stdlib will exist precompiled on a system and just gets linked by the user. This brings me to the question (different issue probably) whether the interfaces of stdlib are bind(C), even if they are intended to be used only in Fortran. But they are many not so unlikely situations where a user would want to use a gfortran-compiled stdlib to be used in a project compiled with the Intel compiler or vice versa.

Reply:
> But they are many not so unlikely situations where a user would want to use a gfortran-compiled stdlib to be used in a project compiled with the Intel compiler or vice versa.

My understanding is that this isn't possible - which is a general issue with Fortran and the module file format (not specifically a limitation of stdlib).

Reply:
@Romendakil So far building stdlib on demand using fpm or the [CMake template](https://github.com/fortran-lang/stdlib-cmake-example) seems the most straight-forward way to use stdlib without having to deal with Fortran ABI issues.

In the future stdlib might be shipped together with compilers and installed in a location specific to the respective compiler.

Reply:
No, that's not true. Of course, you cannot use compiled modules from a different compiler, but we are regularly using in our code third-party libraries that are only available for gfortran and include them as shared libraries in our code compiled with ifort or nagfor.

Reply:
@Romendakil Interesting -- so you can do this by simply having the Fortran interfaces be `bind(C)`?

Reply:
@gareth-nx yes. For some you don't need this, it works even across compilers. But even with `bind(C)` I think you are not totally guaranteed to get always what you expect, as the accompanying C compiler of the two Fortran compilers could be different, and e.g. have different bit sizes of their integer or float data types. However, in general it works. The usual situation is that most of my colleagues in theoretical physics do development only with gfortran, and they have some sloppiness which is tolerated by gfortran. So I cannot compile their code with Nagfor. I do development for our own code mostly with Nagfor, but as the third-party code is bind(C) I can nevertheless link it and use it correctly.

Reply:
I have pushed a commit that edits Makefile.manual as suggested by @awvwgk . Further, in the README there is also explanation of how to specify FC and FFLAGS when building with make.

Recall the purpose is to make the process of compiling with optimization clearer for users, especially novices, while keeping the document brief.